File: typescript/src/run-all.ts
/**
 * Runs the whole walkthrough in order.
 *
 * Read-only steps always run. Write steps announce themselves and skip unless
 * PURECHAIN_DEMO_WRITE=1 is set.
 */

import { writesEnabled } from "./shared.ts";

const steps = [
  ["01", () => import("./01-connect.ts")],
  ["02", () => import("./02-wallet.ts")],
  ["03", () => import("./03-read.ts")],
  ["04", () => import("./04-send.ts")],
  ["05", () => import("./05-contract.ts")],
  ["06", () => import("./06-events.ts")],
  ["07", () => import("./07-metrics.ts")],
] as const;

console.log("purechain-sdk — TypeScript walkthrough");
console.log(writesEnabled() ? "writes ENABLED — this will broadcast transactions" : "read-only (set PURECHAIN_DEMO_WRITE=1 to include write steps)");

for (const [name, load] of steps) {
  try {
    await (await load()).main();
  } catch (err) {
    console.error(`\n  step ${name} failed:`, err instanceof Error ? err.message : err);
    process.exitCode = 1;
  }
}

console.log("\n" + "─".repeat(64));
console.log("done");

File: python/src/run_all.py
"""Runs the whole walkthrough in order.

Read-only steps always run. Write steps announce themselves and skip unless
PURECHAIN_DEMO_WRITE=1 is set.
"""

from __future__ import annotations

import asyncio
import importlib
import sys

from shared import writes_enabled

STEPS = [
    "01_connect",
    "02_wallet",
    "03_read",
    "04_send",
    "05_contract",
    "06_events",
    "07_metrics",
]


async def main() -> None:
    print("purechain-sdk — Python walkthrough")
    print(
        "writes ENABLED — this will broadcast transactions"
        if writes_enabled()
        else "read-only (set PURECHAIN_DEMO_WRITE=1 to include write steps)"
    )

    failed = False
    for name in STEPS:
        try:
            await importlib.import_module(name).main()
        except Exception as err:  # noqa: BLE001 - one step failing should not stop the tour
            print(f"\n  step {name} failed: {err}")
            failed = True

    print("\n" + "─" * 64)
    print("done")
    if failed:
        sys.exit(1)


if __name__ == "__main__":
    asyncio.run(main())
